### ISSUE TITLE
🐛 `useExhaustiveDependencies`: False positive when destructuring props in function body

### ISSUE THREAD

**Issue by maug** [biomejs/biome#8883]
### Environment information

<details>

```bash
CLI:
  Version:                      2.3.12
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.13.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/10.9.2

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         /Users/maug/dev/test/biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    unset

Workspace:
  Open Documents:               0

```

</details>

### What happened?

The `useExhaustiveDependencies` rule incorrectly reports `msg` as an unnecessary dependency when props are destructured in the function body instead of in the parameters.

## Code example

```tsx
import { useEffect } from 'react'

type ExampleProps = {
  msg: string;
};

// ✅ No warning - destructuring in parameters
const _DestructuredInParams: React.FC<ExampleProps> = ({ msg }) => {
  useEffect(() => console.log(msg), [msg]); // correct behavior
};

// ❌ False positive - destructuring in body
const _DestructuredInBody1: React.FC<ExampleProps> = props => {
  const { msg } = props
  useEffect(() => console.log(msg), [msg]); // should NOT trigger useExhaustiveDependencies
};

// ❌ False negative - destructuring in body
const _DestructuredInBody2: React.FC<ExampleProps> = props => {
  const { msg } = props
  useEffect(() => console.log(msg), []); // should trigger useExhaustiveDependencies
};
```

[Playground link](https://biomejs.dev/playground/?code=aQBtAHAAbwByAHQAIAB7ACAAdQBzAGUARQBmAGYAZQBjAHQAIAB9ACAAZgByAG8AbQAgACcAcgBlAGEAYwB0ACcACgAKAHQAeQBwAGUAIABFAHgAYQBtAHAAbABlAFAAcgBvAHAAcwAgAD0AIAB7AAoACQBtAHMAZwA6ACAAcwB0AHIAaQBuAGcAOwAKAH0AOwAKAAoAYwBvAG4AcwB0ACAAXwBEAGUAcwB0AHIAdQBjAHQAdQByAGUAZABJAG4AUABhAHIAYQBtAHMAOgAgAFIAZQBhAGMAdAAuAEYAQwA8AEUAeABhAG0AcABsAGUAUAByAG8AcABzAD4AIAA9ACAAKAB7ACAAbQBzAGcAIAB9ACkAIAA9AD4AIAB7AAoAIAAgAHUAcwBlAEUAZgBmAGUAYwB0ACgAKAApACAAPQA%2BACAAYwBvAG4AcwBvAGwAZQAuAGwAbwBnACgAbQBzAGcAKQAsACAAWwBtAHMAZwBdACkAOwAgAC8ALwAgAGMAbwByAHIAZQBjAHQAIABiAGUAaABhAHYAaQBvAHIACgB9ADsACgAKAGMAbwBuAHMAdAAgAF8ARABlAHMAdAByAHUAYwB0AHUAcgBlAGQASQBuAEIAbwBkAHkAMQA6ACAAUgBlAGEAYwB0AC4ARgBDADwARQB4AGEAbQBwAGwAZQBQAHIAbwBwAHMAPgAgAD0AIABwAHIAbwBwAHMAIAA9AD4AIAB7AAoAIAAgAGMAbwBuAHMAdAAgAHsAIABtAHMAZwAgAH0AIAA9ACAAcAByAG8AcABzAAoAIAAgAHUAcwBlAEUAZgBmAGUAYwB0ACgAKAApACAAPQA%2BACAAYwBvAG4AcwBvAGwAZQAuAGwAbwBnACgAbQBzAGcAKQAsACAAWwBtAHMAZwBdACkAOwAgAC8ALwAgAHMAaABvAHUAbABkACAATgBPAFQAIAB0AHIAaQBnAGcAZQByACAAdQBzAGUARQB4AGgAYQB1AHMAdABpAHYAZQBEAGUAcABlAG4AZABlAG4AYwBpAGUAcwAKAH0AOwAKAAoAYwBvAG4AcwB0ACAAXwBEAGUAcwB0AHIAdQBjAHQAdQByAGUAZABJAG4AQgBvAGQAeQAyADoAIABSAGUAYQBjAHQALgBGAEMAPABFAHgAYQBtAHAAbABlAFAAcgBvAHAAcwA%2BACAAPQAgAHAAcgBvAHAAcwAgAD0APgAgAHsACgAgACAAYwBvAG4AcwB0ACAAewAgAG0AcwBnACAAfQAgAD0AIABwAHIAbwBwAHMACgAgACAAdQBzAGUARQBmAGYAZQBjAHQAKAAoACkAIAA9AD4AIABjAG8AbgBzAG8AbABlAC4AbABvAGcAKABtAHMAZwApACwAIABbAF0AKQA7ACAALwAvACAAcwBoAG8AdQBsAGQAIAB0AHIAaQBnAGcAZQByACAAdQBzAGUARQB4AGgAYQB1AHMAdABpAHYAZQBEAGUAcABlAG4AZABlAG4AYwBpAGUAcwAKAH0AOwAKAA%3D%3D&ruleDomains.react=all)

## Actual behavior

The second component (`_DestructuredInBody1`) triggers:
> This hook specifies more dependencies than necessary: msg

The third component passes as correct.

## Expected behavior

Both components are functionally identical. Neither should produce a warning since `msg` is:
1. Used inside the `useEffect` callback
2. Correctly listed in the dependency array

Third component should trigger `useExhaustiveDependencies`.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
